pkg/azuredisk/controllerserver.go

```diff
@@ -846,6 +854,28 @@ func (d *Driver) getSnapshotByID(ctx context.Context, resourceGroup, snapshotID,
 	return generateCSISnapshot(sourceVolumeID, &snapshot)
 }
 
+// GetSourceDiskSize recursively searches for the sourceDisk and returns: sourceDisk disk size, error
+func (d *Driver) GetSourceDiskSize(ctx context.Context, resourceGroup, diskName string) (*int32, error) {
+	result, rerr := d.cloud.DisksClient.Get(ctx, resourceGroup, diskName)
+	if rerr != nil {
+		return nil, rerr.Error()
+	}
+
+	if result.DiskProperties == nil {
+		return nil, nil
+	}
+
+	if result.DiskProperties.CreationData != nil && (*result.DiskProperties.CreationData).CreateOption == "Copy" {
```

Comment:
is there a dead loop possibility that `CreateOption == "Copy"`, may define a max depth to avoid this issue.